### PR TITLE
Add Mount trait example

### LINKED ISSUE #3140: Mount trait usage example
Provide some example to show how to use the trait. See other examples in https://github.com/apache/camel-k/tree/main/examples/traits as a reference.

### PR DESCRIPTION
**Pull request by haanhvu**
fixes #3140

**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Review by tadayosi — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

**Review by squakez — CHANGES_REQUESTED** [review]
We already have that scenario covered as suggested in the related issue https://github.com/apache/camel-k/tree/main/examples/volumes. I think we need to go beyond the simple timer to log example. One suggestion I can do is to provide some realistic situation. In the volume example, you can see I've created a producer Integration writing and a consumer Integration reading to/from a shared Volume.
You may think on some alternative situation, for example, showing how to log the camel k to an external volume (see https://stackoverflow.com/a/70063203/10974363 for details how to specify log file path). Then you can show how you can scrape that log from another integration.
It's just an idea. I guess the final goal of these activities is to get a bit of knowledge on the platform.

**Comment by squakez** [comment]
I think we still need something. Let me provide here a sort of script the user could run, and you will be able to provide the related Integrations:
```
kamel run Producer.java --volume my-pv-claim:/tmp/log -p quarkus.log.file.path=/tmp/log/trace.log -p quarkus.log.file.enable=true

kamel run Consumer.java --volume my-pv-claim:/tmp/log
```
The Producer could be a timer to log Integration. You can provide some incremental counter, ie, simple language `${exchangeProperty.CamelTimerCounter}`. Ideally the consumer should process the file in order to do something (ie, filtering for some number), so you can have a basic example with a few useful things in it.

**Comment by haanhvu** [comment]
> I think we still need something. Let me provide here a sort of script the user could run, and you will be able to provide the related Integrations:
> 
> ```
> kamel run Producer.java --volume my-pv-claim:/tmp/log -p quarkus.log.file.path=/tmp/log/trace.log -p quarkus.log.file.enable=true
> 
> kamel run Consumer.java --volume my-pv-claim:/tmp/log
> ```
> 
> The Producer could be a timer to log Integration. You can provide some incremental counter, ie, simple language `${exchangeProperty.CamelTimerCounter}`. Ideally the consumer should process the file in order to do something (ie, filtering for some number), so you can have a basic example with a few useful things in it.

@squakez I did this. Please review.

File: examples/traits/mount/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+In this section you will find examples about fine tuning your `Integration` using **Mount** `trait` capability.
+
+
+**Example 1: Mount Persistent Volume Claim on the Integration Pod in a Kubernetes platform (Static Volume Provisioning)**
+
+Requirement: Persistent Volume already created.
+
+Create the Pod and the Persistent Volume Claim:
+`kubectl apply -f pod-example.yaml`
```
Comment (squakez) [inline examples/traits/mount/README.md:12]: You don't need any live Pod for the PVC to be usable. You can just remove this line.

File: examples/traits/mount/pod-example.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
Comment (squakez) [inline examples/traits/mount/pod-example.yaml:1]: This yaml is not needed, you can remove it.

**Review by squakez — APPROVED** [review]
Good work, thanks! Just remove the references of the Pod creation which are superfluous.

File: examples/traits/mount/README.md (hunk 1 above)
Comment (haanhvu) [reply examples/traits/mount/README.md]: done!

**Review by squakez — APPROVED** [review]

**Comment by haanhvu** [comment]
@squakez can this be merged now?

**Pull request merged**